compiler/ballerina-backend-jvm/src/main/ballerina/compiler_backend_jvm/jvm_cast_gen.bal

```diff
@@ -28,6 +28,9 @@ function generateCheckCast(jvm:MethodVisitor mv, bir:BType sourceType, bir:BType
     } else if (targetType is bir:BTypeString) {
         generateCheckCastToString(mv, sourceType);
         return;
+    } else if (targetType is bir:BTypeDecimal) {
+        checkCast(mv, targetType);
```

Comment:
I think we need to do something similar what we have done for other value-types

Reply:
Also need to add a similar if-case to `generateCast()` as well

Reply:
fixed.